### DISCUSSION TITLE
Changing operation names for handlers in the schema

### DISCUSSION THREAD

**Discussion by scott2b** [litestar-org/litestar#1361] [Q&A] [answered]
Maybe I am looking in the wrong place for this functionality. By default, HTTP-method decorated route handlers are given title-case converted names in the schema documentation. For some handlers, the title casing is not ideal, or otherwise one might want a name that is not necessarily the same as the name of the function.

My expectation would be that the name parameter passed to the decorator would override the default name, but that does not seem to be the case. Is there a way to change the operation name in the docs for an operation associated with a handler?

e.g., I am expecting this would override "Urls" in the docs to being "URLs":

```
@get(path="/urls", name="URLs")
def urls( ...
```

**Comment by scott2b** [comment, marked as answer]
Oh, I found it. The parameter is `operation_id`. So, for future reference, this should work:

```
@get(path="/urls", operation_id="URLs")
def urls( ...
```

**Reply by JacobCoffee** [reply]
@scott2b Was this hard to find in the docs or unclear in some way? Just curious if this could use a facelift on the documentation side

**Reply by scott2b** [reply]
Thanks for asking. I think this mostly boils down to my ignorance of the OpenAPI specification. `operationId` is the part of that specification that I was not really aware of until grappling with this problem. Once I understood that the thing being labeled in, e.g. redoc is an "operation" and that what I perceived as a name or label was in fact the specified operationId, the parameter finally jumped out at me in the Starlite API docs.

It might be useful to have something in the usage docs (perhaps I missed it), but I think the API documentation is fine.
